src/test/java/seedu/storemando/logic/commands/EditCommandTest.java

```diff
@@ -118,6 +112,53 @@ public void execute_duplicateItemFilteredList_failure() {
         assertCommandFailure(editCommand, model, EditCommand.MESSAGE_DUPLICATE_ITEM);
     }
 
+    @Test
+    public void execute_allSameFieldsSpecifiedUnfilteredList_failure() {
+        Item firstItem = model.getFilteredItemList().get(INDEX_FIRST_ITEM.getZeroBased());
+        EditCommand.EditItemDescriptor descriptor = new EditItemDescriptorBuilder(firstItem).build();
+        EditCommand editCommand = new EditCommand(INDEX_FIRST_ITEM, descriptor);
+        assertCommandFailure(editCommand, model, EditCommand.MESSAGE_NO_CHANGE);
+    }
+
+    @Test
+    public void execute_someSameFieldsSpecifiedUnfilteredList_failure() {
+        Index indexLastItem = Index.fromOneBased(model.getFilteredItemList().size());
+        Item lastItem = model.getFilteredItemList().get(indexLastItem.getZeroBased());
+
+        EditCommand.EditItemDescriptor descriptor = new EditItemDescriptorBuilder()
+            .withName(lastItem.getItemName().toString()).build();
+        EditCommand editCommand = new EditCommand(indexLastItem, descriptor);
+
+        String expectedMessage = String.format(EditCommand.MESSAGE_NO_CHANGE);
+
+        assertCommandFailure(editCommand, model, expectedMessage);
+    }
+
+    @Test
+    public void execute_allSameFieldsSpecifiedFilteredList_failure() {
+        showItemAtIndex(model, INDEX_FIRST_ITEM);
+
```

Comment:
Maybe you can use StoreMando instead of storemando for consistency

Reply:
We can leave it as it is for now as mentioned above.